### PR TITLE
MSTEST0013: Flag cases where assemblycleanup is set on generic test class

### PR DESCRIPTION
**Pull request by engyebrahim**
fix: https://github.com/microsoft/testfx/issues/2421

### CODE REVIEW

**Review by MarcoRossignoli — APPROVED** [review]

File: src/Analyzers/MSTest.Analyzers/AssemblyCleanupShouldBeValidAnalyzer.cs [hunk 1]
```diff
@@ -61,6 +62,8 @@ public override void Initialize(AnalysisContext context)
         INamedTypeSymbol? valueTaskSymbol, bool canDiscoverInternals)
     {
         var methodSymbol = (IMethodSymbol)context.Symbol;
+        var containingTypeSymbol = context.Symbol.ContainingType;
```
Comment (Evangelink) [inline src/Analyzers/MSTest.Analyzers/AssemblyCleanupShouldBeValidAnalyzer.cs:65]: I would move this variable close to where it's used. Also, I would probably just drop it and use directly `context.Symbol.ContainingType.IsGenericType` on line 80.

File: src/Analyzers/MSTest.Analyzers/Resources.resx [hunk 2]
```diff
@@ -130,6 +130,9 @@
   <data name="AssemblyCleanupShouldBeValidMessageFormat_NoParameters" xml:space="preserve">
     <value>AssemblyCleanup method '{0}' should not take any parameter</value>
   </data>
+  <data name="AssemblyCleanupShouldBeValidMessageFormat_NotAGenericClass" xml:space="preserve">
+    <value>AssemblyCleanup method '{0}' can't be inside a generic class</value>
```
Comment (Evangelink) [inline src/Analyzers/MSTest.Analyzers/Resources.resx:134]: Maybe `AssemblyCleanup method '{0}' can't be declared on a generic class`

**Review by Evangelink — COMMENTED** [review]
A few nits, that would be good to apply them to the other rules as I expect they are done similarly.

cc @MarcoRossignoli

**Comment by MarcoRossignoli** [comment]
@engyebrahim pls follow up the @Evangelink comments

**Pull request merged**